### ISSUE TITLE
Dashboard: iHPC app developer can't override smart attribute widget

### ISSUE THREAD

**Issue by nickjer** [OSC/ondemand#223]
A developer is unable to specify a `form.yml` with:

```yaml
cluster: cluster1
attributes:
  bc_account:
    widget: select
    options:
      - [ "q1", "q1" ]
      - [ "q2", "q2" ]
form:
  - bc_account
```

where the developer wants the smart attribute `bc_account` to be a selection field.

As the code will always set this as a text field:

https://github.com/OSC/ood-dashboard/blob/a009cf3ce9f108ccac4cf301fd2ee821214bd564/lib/smart_attributes/attributes/bc_account.rb#L15-L17

**Comment by ericfranz** [comment]
https://discourse.osc.edu/t/cant-change-widget-type-of-pre-defined-form-elements/1000

**Comment by johrstrom** [comment]
This was fixed in #1482.
